src/tests/test_types.cpp

```diff
@@ -264,4 +264,72 @@ INSTANTIATE_TEST_SUITE_P(TestStringSuite, TestString,
                                            TestStringParam("abc", ""),
                                            TestStringParam("bb", "aa")));
 
-}  // namespace tests
+struct TestDateTypeParam {
+    std::string const l;
+    std::string const r;
+
+    TestDateTypeParam(std::string l, std::string r) noexcept : l(std::move(l)), r(std::move(r)) {}
+};
+
+class TestDateType : public ::testing::TestWithParam<TestDateTypeParam> {};
+
+TEST_P(TestDateType, ComparisonTest) {
+    TestDateTypeParam p = GetParam();
+    auto const& [l_val, r_val] = GetParam();
+    std::unique_ptr<mo::Type> type(mo::CreateType(mo::TypeId::kDate, true));
+    auto* date_type = static_cast<mo::DateType*>(type.get());
+    mo::Date d1(boost::gregorian::from_string(l_val));
+    mo::Date d2(boost::gregorian::from_string(r_val));
+    model::CompareResult cr = mo::DateType::Compare(d1, d2);
+
+    using Owner = std::unique_ptr<std::byte[]>;
+    Owner l_owner(date_type->MakeValue(d1));
+    Owner r_owner(date_type->MakeValue(d2));
+    std::byte* l = l_owner.get();
+    std::byte* r = r_owner.get();
+    EXPECT_EQ(date_type->Compare(l, r), cr);
```

Comment:
Given that `date_type->Compare` uses `mo::DateType::Compare` in its implementation, this check seems pretty useless because it will almost never fail